### ISSUE TITLE
test solr backup restore

### ISSUE THREAD

**Issue by hroberts** [berkmancenter/mediacloud#31]
The solr backup finished last night.  It took 7.5 hours.

We should now have a read consistent copy of the solr database, but we need to test to make sure that we can actually restore the cluster.  I think the simplest way to do so is to fire up three ec2 instances and try to get the system running on them. 

Solrcloud's mapping of server:port combinations to shards is a blackbox (to me at least), so sadly I don't think we can test by just opening them on one server, at least without having to fiddle a bunch with the network setup.

Also, for reasons I have never understood, one of the shards on mcquery2 always gets mapped in an ip address rather than a hostname, so I think we'll have to spoof an ip address.

**Comment by pypt** [comment]
Started uploading 2 TB worth of Solr dump to S3. When finished, will start three EC2 instances, copy shard data to them and try out the backup.

**Comment by pypt** [comment]
Uploaded dump to S3, starting to copy to EC2 machines.

**Comment by pypt** [comment]
Copied backups from S3 to three EC2 instances that are currently running, will try to join them all into a cluster next thing tomorrow.

**Comment by pypt** [comment]
On my first attempt, I screwed up the hostnames a little (used `test-mcqueryX` instead of `mcqueryX`), but the cluster started more or less normally, except that it tried to join production's `mcquery2.media.mit.edu` (`18.85.22.24:8983`) to the cluster:

![solr-test-dump-replicas](https://cloud.githubusercontent.com/assets/181949/16536280/55a12b86-3ff8-11e6-854b-3512abd4d195.png)

(`10.179.181.232` is test `mcquery2`.)

After setting hostnames from `test-mcqueryX` to `mcqueryX`, the test cluster somehow managed to convince live `mcquery2` to join as replica:

![solr-test-dump-replicas-new](https://cloud.githubusercontent.com/assets/181949/16537330/b5555452-4008-11e6-8d27-9dcd8a3a15d8.png)

I've tried editing shard hosts manually in ZooKeeper's `/clusterstate.json` but that didn't help. I also noticed uncommitted `clusterstate.json` and `foo` in backup's directories, so I can only assume that you have either participated in this hostname hardcoding initiative or tried to fix it :)

My conclusion would be that the backup itself is fine (querying works too), it's cluster names / hosts that are a little messed up in a way that the cluster now works properly only from a specific location (live `mcqueryX` servers).

Terminating the test instances, removing backup from S3, will close this issue afterwards.
